Set baudrate from metadata

darkxst opened the pull request:
Use the metadata to set the baudrate for the build rather than individual patches.

This is useful when building a variety of baudrates for the same device.

agners commented on .github/workflows/silabs-firmware-build.yaml:
```diff
@@ -64,6 +64,16 @@ jobs:
+      - name: Set Baudrate
+        env:
+          baudrate: ${{ fromJSON(inputs.metadata_extra).baudrate }}
+        if: ${{ env.baudrate != 115200 }}
```
Hm, what if their default changes? I mean this is essentially a nop, I think I'd rather suggest to avoid that risk and drop this if... Thoughts?

darkxst commented:
Yes, good point. I hadn't really considered that. I suppose its also possible baudrate may have been changed via the .slcp file in some cases.

I have dropped the if, so baudrate will always be consistent with whatever is set in the metadata.

Hedda commented:
> Hm, what if their default changes?

Noticed that puddly added baudrates in an object to manifest.json file for sl-web-tools -> https://github.com/NabuCasa/sl-web-tools/pull/7

agners commented:
> FYI, noticed that @puddly also added baudrates in an object to manifest.json file for sl-web-tools -> [NabuCasa/sl-web-tools#7](https://github.com/NabuCasa/sl-web-tools/pull/7)

Actually, that brings up a good point: We should make the baudrate a proper input in the `silabs-firmware-build.yaml`, and add it to every builds metadata (in 
https://github.com/NabuCasa/silabs-firmware-builder/blob/main/.github/workflows/silabs-firmware-build.yaml#L102).

This then would allow to access the baudrate easier for this patch.

agners commented:
> CPC sooner rather than later.

For CPS multiprotocol fimrware we already changed to 460800 a week ago or so.

darkxst commented:
> FYI, noticed that @puddly also added baudrates in an object to manifest.json file for sl-web-tools -> [NabuCasa/sl-web-tools#7](https://github.com/NabuCasa/sl-web-tools/pull/7)

The baudrates in the manifest provides the list of baudrates that should be probed by universal-silabs-flasher when seaching for installed firmware. 

> We should make the baudrate a proper input

OK I will split it out and make it a `required` input. That actually works well for my use case as I plan to make baudrate an input to the matrix.

> For CPS multiprotocol fimrware we already changed to 460800 a week ago or so.

This change was already covered ;) It only would have been a problem if silabs changed their defaults. Anyway now the baudrate will be updated regardless of what is set by silabs or the .slcp file.
